client/packages/common/src/hooks/useNativeClient/useBackButtonHandler.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { App } from '@capacitor/app';
+import { useNavigate } from '@openmsupply-client/common';
+import { Capacitor } from '@capacitor/core';
+
+export const useBackButtonHandler = ({
+  isNavigateEnabled,
+}: {
+  isNavigateEnabled: boolean;
+}) => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (Capacitor.isNativePlatform()) {
+      App.removeAllListeners();
+
+      App.addListener('backButton', ({ canGoBack }) => {
+        if (canGoBack && isNavigateEnabled) navigate(-1);
+        else App.exitApp();
+      });
+    }
+
+    return () => {
+      if (Capacitor.isNativePlatform()) {
+        App.removeAllListeners();
```

Comment:
I did play around with specific handles but have had so much trouble getting this tested, I've given up and removed them all.

The `addListener` call does return a promise that holds an object with a `remove` function, so I assigned that to a ref and called that. It might still work, but as I say, I've spent long enough on this.

Reply:
Nit return the clean up method only in `if (Capacitor.isNativePlatform()) { }`

Please ignore me if you think this would require further testing...